server/status.go

```diff
@@ -208,10 +209,37 @@ func (s *statusServer) handleNodesStatus(w http.ResponseWriter, r *http.Request,
 
 // handleNodeStatus handles GET requests for a single node's status. If no id is
 // available, it calls handleNodesStatus to return all node's statuses.
-func (s *statusServer) handleNodeStatus(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	// TODO(bram) parse node-id in path and return the single node's status
-	// only.
-	s.handleNodesStatus(w, r, nil)
+func (s *statusServer) handleNodeStatus(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	id, err := strconv.ParseInt(ps.ByName("id"), 10, 64)
+	if err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	key := engine.NodeStatusKey(int32(id))
+
+	nodeStatus := &proto.NodeStatus{}
+	call := client.GetProto(key, nodeStatus)
+	resp := call.Reply.(*proto.GetResponse)
+	if err := s.db.Run(call); err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if resp.Error != nil {
```

Comment:
Or maybe not.
in the snippet below from `Run`, `err` is a named return. That seems to corroborate my initial comment.
```
		kv.Sender.Send(c)
		err = c.Reply.Header().GoError()
		if err != nil {
			log.Infof("failed %s: %s", c.Method(), err)
		} else if c.Post != nil {
			err = c.Post()
		}
		return
```